src/java/qfs-access/src/main/java/com/quantcast/qfs/access/KfsTest.java

```diff
@@ -503,4 +506,46 @@ private static void verifyFileAttr(KfsAccess kfsAccess,
             System.exit(1);
         }
     }
+
+    private static void testDisableReadAhead(KfsAccess kfsAccess, String baseDir)
+            throws IOException {
+        String filePath = baseDir + "/sample_file.1";
+        String createParams = "S";
+        KfsOutputChannel outputChannel = kfsAccess.kfs_create_ex(filePath,
+                true, createParams);
+        int numBytes = 1048576;
+        char[] dataBuf = new char[numBytes];
+        generateData(dataBuf, numBytes);
+        String s = new String(dataBuf);
+        byte[] buf = s.getBytes();
+        ByteBuffer b = ByteBuffer.wrap(buf, 0, buf.length);
+        int res = outputChannel.write(b);
+        if (res != buf.length) {
+            System.out.println("Was able to write only: " + res);
+            System.exit(1);
+        }
+        if (outputChannel == null) {
+            System.out.println("Unable to call create");
+            System.exit(1);
+        }
+        outputChannel.sync();
+        outputChannel.close();
+
+        KfsInputChannel inputChannel = kfsAccess.kfs_open(filePath);
+        if (inputChannel == null) {
+            System.out.println("open on " + filePath + "failed!");
+            System.exit(1);
+        }
+        inputChannel.setReadAheadSize(0);
+
+        byte[] dstBuf = new byte[128];
+        res = inputChannel.read(ByteBuffer.wrap(dstBuf, 0, 128));
```

Comment:
Issuing single read doesn't seem that interesting. Better cover coverage at low cost: small read seek forward (skip), and possible seek back.